### PR TITLE
Yuqiang/teacher license code

### PR DESCRIPTION
**Pull request by smallst**
add an `teacher license code` link in /admin, which can generate codes
图片

add an `/teachers/licenses/join?codes=${codes}` url to join by code （since nick said this function is rarely used, i haven't add a significant UI entry for it.)
 - codes is a string of code join by `,` , for example `T-85NI-WJSG-U4OD,T-85NI-HRGC-U4OD`

teacher can see his new license in licenseView like below (shared from admin):
图片

### CODE REVIEW

File: app/views/admin/components/license/LicenseType.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<script>
+  const utils = require('core/utils')
```
[the remark this answers is not in the document]
Comment (smallst) [reply app/views/admin/components/license/LicenseType.vue:2]: okay. seems this branch didn't has this hooks. i can merge latest master

File: app/views/teachers/JoinLicensesByCode.vue [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      }
+    }
+  }
+</script>
+
+<template lang="pug">
+  div
+    .progress.progress-striped.active(v-if="state === 'loading'")
+      .progress-bar(style="width: 100%")
+    .alert.alert-success(v-if="state === 'success'") Licenses have been Activated Successfully!
```
[the remark this answers is not in the document]
Comment (smallst) [reply app/views/teachers/JoinLicensesByCode.vue:48]: oh good point

File: app/views/teachers/JoinLicensesByCode.vue [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          if(state == 'success') {
+            setTimeout(() => {
+              application.router.navigate('/teachers/licenses', {trigger: true})
+            }), 3000
+          }
+        })
+      }
+    },
+    mounted () {
+      if(this.$route.query.codes) {
```
[the remark this answers is not in the document]
Comment (smallst) [reply app/views/teachers/JoinLicensesByCode.vue:35]: yeah, now the partner would only use 3 or 5 codes at the same time. we can update it later.

File: app/views/teachers/JoinLicensesByCode.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return {
+      state: 'loading',
+      result: {}
+    }
+  },
+  mounted () {
+    if (this.$route.query.codes) {
+      this.post()
+    } else {
+      application.router.navigate('/teachers/licenses', { trigger: true })
```
Comment (mrfinch) [inline app/views/teachers/JoinLicensesByCode.vue:14]: instead of this, just showing error message on this page would be better experience.

**Pull request merged**